darshan-util/pydarshan/tests/test_cffi_misc.py

```diff
@@ -62,3 +63,97 @@ def test_file_hash_type(log_path):
+            3.936579942703247, 0.0, 115.0781660079956, 115.77035808563232,
+            0.0, 100397.60042190552, 11.300841808319092, 0.0,
+            17.940945863723755, 20.436099529266357, 85.47495031356812,
+            0.0, 0.0,
+        ]
+    )
+
+    # generate a record from sample log
+    log = backend.log_open("tests/input/sample.darshan")
+    rec = backend.log_get_generic_record(log=log, mod_name="POSIX", dtype=dtype)
+
+    # each record should have the following keys
+    record_keys = ["id", "rank", "counters", "fcounters"]
+    assert list(rec.keys()) == record_keys
+    # check the file hash/id
+    assert rec["id"] == 6301063301082038805
+    # check the rank
+    assert rec["rank"] == -1
+
+    if dtype == "numpy":
+        # check the length of the returned arrays are correct
+        assert rec["counters"].size == 69
+        assert rec["fcounters"].size == 17
+        # collect the actual counter/fcounter values
+        actual_counter_vals = rec["counters"]
+        actual_fcounter_vals = rec["fcounters"]
+
+    elif dtype == "dict":
+        # check the length of the returned dictionaries are correct
+        assert len(rec["counters"]) == 69
+        assert len(rec["fcounters"]) == 17
+        # collect the actual counter/fcounter key names
+        actual_counter_names = list(rec["counters"].keys())
+        actual_fcounter_names = list(rec["fcounters"].keys())
+        # collect the actual counter/fcounter values
+        actual_counter_vals = np.asarray(list(rec["counters"].values()))
+        actual_fcounter_vals = np.asarray(list(rec["fcounters"].values()))
+
+    elif dtype == "pandas":
+        # make sure the added column keys are in the dataframes
+        for key in ["id", "rank"]:
+            assert key in rec["counters"].columns
+            assert key in rec["fcounters"].columns
+        # double check the id/rank values
+        assert rec["counters"]["id"].values == 6301063301082038805
+        assert rec["counters"]["rank"].values == -1
+        # make sure the dataframes are the expected shapes
+        # the shapes are 2 larger than the arrays since the id/rank
+        # columns are added to the dataframes
+        assert rec["counters"].shape == (1, 71)
+        assert rec["fcounters"].shape == (1, 19)
+        # collect the actual counter/fcounter key names
+        # don't include the id/rank columns
+        actual_counter_names = list(rec["counters"].columns)[2:]
+        actual_fcounter_names = list(rec["fcounters"].columns)[2:]
+        # collect the actual counter/fcounter values
+        # don't include the id/rank columns
+        actual_counter_vals = rec["counters"].values[0][2:]
+        actual_fcounter_vals = rec["fcounters"].values[0][2:]
+
+    # check the actual counter/fcounter values agree
+    # with the expected counter/fcounter values
+    assert_array_equal(expected_counter_vals, actual_counter_vals)
+    assert_allclose(expected_fcounter_vals, actual_fcounter_vals)
```

Comment:
If you read the documentation for assert_allclose et al. you'll see that you've got your parameters backwards here--please fix in a follow-up... (actual comes first, then desired/expected)

![image](https://user-images.githubusercontent.com/7903078/129826715-e5ef5e42-1123-4e34-9a28-7f65c86a9e7e.png)